File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/validation/annotation/DateTimeRangeGroup.java
package com.jmsoftware.maf.springcloudstarter.validation.annotation;

import java.lang.annotation.*;

/**
 * Description: DateTimeRangeGroup, change description here.
 * <p>
 * TODO: to support nullable value
 *
 * @author 钟俊（zhongjun）, email: zhongjun@toguide.cn, date: 6/3/2021 2:58 PM
 **/
@Documented
@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface DateTimeRangeGroup {
    String groupName() default "defaultDateTimeRangeGroup";

    DateTimeRangeType type() default DateTimeRangeType.START_TIME;
}

File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/validation/annotation/DateTimeRangeGroup.kt
package com.jmsoftware.maf.springcloudstarter.validation.annotation


/**
 * # DateTimeRangeGroup
 *
 * Description: DateTimeRangeGroup, change description here.
 *
 * TODO: to support nullable value
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 9:53 AM
 */
@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class DateTimeRangeGroup(
    val groupName: String = "defaultDateTimeRangeGroup",
    val type: DateTimeRangeType = DateTimeRangeType.START_TIME
)
